### PR TITLE
[Scripts] Basketball hoop

### PR DESCRIPTION
**Pull request by imgntn**
This script adds an example that spawns a basketball hoop with a working collision hull for shooting baskets.

### CODE REVIEW

File: examples/toys/basketball_hoop/createHoop.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+var DISTANCE_IN_FRONT_OF_ME = 1.0;
+
+var hoopStartPosition =
+    Vec3.sum(MyAvatar.position,
+        Vec3.multiplyQbyV(MyAvatar.orientation, {
+            x: 0,
+            y: 0.0,
+            z: -DISTANCE_IN_FRONT_OF_ME * 2
```
Comment (ZappoMan) [inline examples/toys/basketball_hoop/createHoop.js]: coding standard: 2 is a magic number here..why not just make DISTANCE_IN_FRONT_OF_ME 2 to begin with

File: examples/toys/basketball_hoop/createHoop.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  This is a script that creates a persistent basketball hoop with a working collision hull.  Feel free to move it.
+//  Run basketball.js to make a basketball.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+/*global MyAvatar, Entities, AnimationCache, SoundCache, Scene, Camera, Overlays, HMD, AvatarList, AvatarManager, Controller, UndoStack, Window, Account, GlobalServices, Script, ScriptDiscoveryService, LODManager, Menu, Vec3, Quat, AudioDevice, Paths, Clipboard, Settings, XMLHttpRequest, randFloat, randInt */
+Script.include("../../utilities.js");
```
Comment (ZappoMan) [inline examples/toys/basketball_hoop/createHoop.js]: do you really need these?

File: examples/toys/basketball_hoop/createHoop.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Script.include("../../libraries/utils.js");
```
Comment (ZappoMan) [inline examples/toys/basketball_hoop/createHoop.js]: do you really need these?

**Comment by imgntn** [comment]
good points

**Pull request merged**